hivemind/server/layers/custom_experts.py

```diff
@@ -0,0 +1,31 @@
+import os
+import importlib
+import torch
+import torch.nn as nn
+from typing import Callable, Type
+
+from hivemind.server.layers import name_to_block, name_to_input
+
+def add_custom_models_from_file(path: str):
+    spec = importlib.util.spec_from_file_location(
+        "custom_module", os.path.abspath(path))
+    foo = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(foo)
+
+def register_expert_class(name: str, sample_input: Callable[[int, int], torch.tensor]):
+    """
+    Adds a custom user expert to hivemind server.
+    :param name: the name of the expert. It shouldn't coincide with existing modules\
+        ('ffn', 'transformer', 'nop', 'det_dropout')
+    :param sample_input: a function which gets batch_size and hid_dim and outputs a \
+        sample of an input in the module
+    :unchanged module
+    """
+    def registrator(custom_class: Type[nn.Module]):
```

Comment:
It's a function, not a class/entity, so it's probably best to rename this

Reply:
e.g. register_expert

Reply:
renamed